public/src/components/channelManagement/campaigns/CampaignsForm.tsx

```diff
@@ -100,21 +118,25 @@ const CampaignsForm: React.FC = () => {
     }
   };
 
-  const currentCampaign = getSelectedCampaign();
+  const hasCampaignBeenSelected = checkForSelectedOrRequestedCampaign();
```

Comment:
A few issues here -
1. `hasCampaignBeenSelected` is redundant, as we already have `selectedCampaign`.
2. The use of the `useParams` hook in the `checkForSelectedOrRequestedCampaign` function doesn't feel very idiomatic and worries me. Instead, I think `useParams` should be declared at the top with the `useState`s. The `checkForSelectedOrRequestedCampaign` function can then be removed, and the logic to `setSelectedCampaign` can be moved into the `useEffect` at the top (after the campaigns have been retrieved).
3. `testName` should be `campaignName`

Happy to pair on this!

Reply:
Fixed as suggested

Reply:
Nice! If you update the parameter name [here](https://github.com/guardian/support-admin-console/blob/main/public/src/main.tsx#L230) then there's no need for `testName` in this file

Reply:
Done!